lib/install-nix.sh

```diff
@@ -18,19 +18,17 @@ fi
 installer_options=(
   --daemon
   --daemon-user-count 4
+  --no-channel-add
   --darwin-use-unencrypted-nix-store-volume
   --nix-extra-conf-file /tmp/nix.conf
 )
 
-if [[ $INPUT_SKIP_ADDING_NIXPKGS_CHANNEL = "true" || $INPUT_NIX_PATH != "" ]]; then
-  installer_options+=(--no-channel-add)
-else
-  INPUT_NIX_PATH="/nix/var/nix/profiles/per-user/root/channels"
+# On self-hosted runners we don't need to install more than once
+if [ ! -d "/nix/store" ] 
```

Comment:
it's a good idea to always use `[[ ]]` as it has better semantic in terms of argument escaping and consider `[ ]` legacy.
```suggestion
if [[ ! -d /nix/store ]] 
```